ocsmesh/utils.py

```diff
@@ -191,86 +193,14 @@ def get_boundary_segments(mesh):
 
 
 def get_mesh_polygons(mesh):
+    elm_polys = []
+    for elm_type in ELEM_2D_TYPES:
+        elems = getattr(mesh, elm_type)['index']
+        elm_polys.extend(
+            [Polygon(mesh.vert2['coord'][cell]) for cell in elems]
+        )
```

Comment:
Old polygon from mesh **sometimes** failed randomly for cases where two separate polygons are touching at multiple forms, forming *islands* between them

Reply:
This method might be slower and/or more memory hungry (not tested) but it is correct!